pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -523,6 +524,97 @@ def reset(self):
         self._opt_idx: Union[int, None] = None
         self._batch_size: Union[int, None] = None
         self._has_batch_loop_finished = False
+        self.legacy_batch_log_metrics = {}
+        self.legacy_batch_pbar_metrics = {}
+
+    def __call__(
+            self,
+            fx_name: Optional[Union[str, int]] = None,
+            dl_idx: Optional[Union[str, int]] = None,
+            opt_idx: Optional[Union[str, int]] = None,
+            batch_idx: Optional[Union[str, int]] = None,
+            split_idx: Optional[Union[str, int]] = None,
+            reduced: bool = False,
+    ):
+        """
+        This function is an helper to access stored data
+
+        It access data from the HookResultStore. Please,
+        check its data structure for better understanding
+
+        Data can be accessed with the following chains:
+
+        IF REDUCED:
+            * IF accessing a fx_name defined in batch training loop:
+                fx_name -> dl_idx -> opt_idx -> batch_idx -> split_idx
+            * ELSE fx_name -> dl_idx -> batch_idx
+        ELSE:
+            * IF accessing a fx_name defined in batch training loop:
+                fx_name -> dl_idx -> opt_idx
+            * ELSE fx_name -> dl_idx
+
+        Note: As soon as a param is None, it breaks the chain and return associated stored data.
+
+        Example:
+
+        result: Result = self(fx_name="training_step", dl_idx="0", opt_idx="0", reduced=True)
+        result['train_loss_epoch'] # aggregated train_loss over one epoch.
+
+        Args:
+
+            fx_name: Hook name from ModelHooks or Callback. Example: `training_step`
+
+            dl_idx: Dataloader idx in short. It starts from 0 to num_dataloaders - 1
+
+            opt_idx: Optimizer idx in short. It starts from 0 to num_optimizers - 1
+
+            batch_idx: Index of batch idx seen during batch training or evaluation.
+                Works only with reduced=False
+
+            split_idx: Index of split idx in training loop when ttbt is used.
+
+            reduced: Data are being aggregated on on_epoch_end.
+                Indicates if we want to access aggregated Result or not.
+        """
+
+        hook_result = self[str(fx_name)]
+
+        dl_idx = str(dl_idx) if dl_idx is not None else None
+        opt_idx = str(opt_idx) if opt_idx is not None else None
+        batch_idx = str(batch_idx) if batch_idx is not None else None
+        split_idx = int(split_idx) if split_idx is not None else None
+
+        internal_type = hook_result._internal_type
+        if internal_type is None:
+            return Result()
+
+        if reduced:
+            result = hook_result._internals_reduced
+        else:
+            result = hook_result._internals
+
+        if internal_type == ResultStoreType.INSIDE_BATCH_TRAIN_LOOP:
+            if not reduced:
+                if dl_idx is not None:
```

Comment:
This is quite a lot of repeat manual logic, can we not just make this something like an unwrap function?

Reply:
Let's take care of that later if needed. This function isn't really used outside from debugging.

Reply:
And it is pretty clear what is happening for anyone reading it :)